File: frontend/src/api/nav.ts
import api from './index'

export interface NavCategory {
  id: number
  name: string
  icon?: string
  sort_order: number
  created_at: string
}

export interface NavSite {
  id: number
  category_id: number
  name: string
  url: string
  icon?: string
  description?: string
  sort_order: number
  created_at: string
  category?: NavCategory
}

export interface NavStats {
  total_categories: number
  total_sites: number
}

export const navApi = {
  listCategories() {
    return api.get<any, NavCategory[]>('/api/nav/categories')
  },
  createCategory(data: { name: string; icon?: string; sort_order?: number }) {
    return api.post<any, NavCategory>('/api/nav/categories', data)
  },
  updateCategory(id: number, data: { name?: string; icon?: string; sort_order?: number }) {
    return api.put<any, NavCategory>(`/api/nav/categories/${id}`, data)
  },
  deleteCategory(id: number) {
    return api.delete(`/api/nav/categories/${id}`)
  },
  listSites(categoryId?: number) {
    return api.get<any, { total: number; items: NavSite[] }>('/api/nav/sites', {
      params: categoryId ? { category_id: categoryId } : {}
    })
  },
  createSite(data: { category_id: number; name: string; url: string; icon?: string; description?: string; sort_order?: number }) {
    return api.post<any, NavSite>('/api/nav/sites', data)
  },
  updateSite(id: number, data: { category_id?: number; name?: string; url?: string; icon?: string; description?: string; sort_order?: number }) {
    return api.put<any, NavSite>(`/api/nav/sites/${id}`, data)
  },
  deleteSite(id: number) {
    return api.delete(`/api/nav/sites/${id}`)
  },
  getStats() {
    return api.get<any, NavStats>('/api/nav/stats')
  }
}

File: backend/app/schemas/nav.py
"""Navigation module schemas."""

from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List


# ── Category schemas ──────────────────────────────────────────


class NavCategoryCreate(BaseModel):
    name: str
    icon: Optional[str] = None
    sort_order: int = 0


class NavCategoryUpdate(BaseModel):
    name: Optional[str] = None
    icon: Optional[str] = None
    sort_order: Optional[int] = None


class NavCategoryResponse(BaseModel):
    id: int
    name: str
    icon: Optional[str]
    sort_order: int
    created_at: datetime

    class Config:
        from_attributes = True


# ── Site schemas ──────────────────────────────────────────────


class NavSiteCreate(BaseModel):
    category_id: int
    name: str
    url: str
    icon: Optional[str] = None
    description: Optional[str] = None
    sort_order: int = 0


class NavSiteUpdate(BaseModel):
    category_id: Optional[int] = None
    name: Optional[str] = None
    url: Optional[str] = None
    icon: Optional[str] = None
    description: Optional[str] = None
    sort_order: Optional[int] = None


class NavSiteResponse(BaseModel):
    id: int
    category_id: int
    name: str
    url: str
    icon: Optional[str]
    description: Optional[str]
    sort_order: int
    created_at: datetime

    class Config:
        from_attributes = True


# ── List / stats schemas ─────────────────────────────────────


class NavSiteListResponse(BaseModel):
    total: int
    items: List[NavSiteResponse]


class NavStatsResponse(BaseModel):
    total_categories: int
    total_sites: int
